stride/physics/iso_elastic/devito.py

```diff
@@ -0,0 +1,398 @@
+
+import devito
+import numpy as np
+import scipy.signal
+
+import mosaic
+from mosaic.utils import camel_case
+
+from ..common.devito import GridDevito, OperatorDevito
+from ..problem_type import ProblemTypeBase
+from .. import boundaries
+
+
+__all__ = ['IsoElasticDevito']
+
+
+@mosaic.tessera
+class IsoElasticDevito(ProblemTypeBase):
+    """
+
+    Parameters
+    ----------
+    name : str, optional
+        Name of the PDE, defaults to an automatic name.
+    grid : Grid, optional
+        Existing grid, if not provided one will be created. Either a grid or
+        space, time and slow_time need to be provided.
+    space : Space, optional
+    time : Time, optional
+    slow_time : SlowTime, optional
+
+    Notes
+    -----
+
+    For forward execution of the PDE, the following parameters can be used:
+
+        wavelets : Traces
+            Source wavelets.
+        vp : ScalarField
+            Compressional (acoustic) speed of sound of the medium, in [m/s].
+        vs : ScalarField
+            Transverse (shear) speed of sound of the medium, in [m/s].
+        rho : ScalarField
+            Density of the medium in [kg/m^3].
+        problem : Problem
+            Sub-problem being solved by the PDE.
+        save_wavefield : bool or int, optional
+            Whether or not to solve the forward wavefield, defaults to True when
+            a gradient is expected, and to False otherwise. An integer number N can
+            also be provided, in which case the last N timesteps of the wavefield
+            will be saved.
+        boundary_type : str, optional
+            Type of boundary for the wave equation (``sponge_boundary_2`` or
+            ``complex_frequency_shift_PML_2``), defaults to ``sponge_boundary_2``.
+        interpolation_type : str, optional
+            Type of source/receiver interpolation (``linear`` or ``hicks``), defaults
+            to ``linear``.
+        attenuation_power : int, optional
+            Power of the attenuation law if attenuation is given (``0`` or ``2``),
+            defaults to ``0``.
+        drp : bool, optional
+            Whether or not to use dispersion-relation preserving coefficients (only
+            available in some versions of Stride). Defaults to False.
+        kernel : str, optional
+            Type of time kernel to use (``OT2`` for 2nd order in time or ``OT4`` for 4th
+            order in time). If not given, it is automatically decided given the time spacing.
+        undersampling : int, optional
+            Amount of undersampling in time when saving the forward wavefield. If not given,
+            it is calculated given the bandwidth.
+
```

Comment:
all parameters from "save_wavefield" onwards are not used in this operator, they should be removed

Reply:
Correction made